src/main/java/net/mcreator/themes/ThemeLoader.java

```diff
@@ -0,0 +1,148 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.themes;
+
+import com.google.gson.Gson;
+import net.mcreator.io.FileIO;
+import net.mcreator.plugin.PluginLoader;
+import net.mcreator.preferences.PreferencesManager;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.laf.MCreatorTheme;
+import net.mcreator.util.image.ImageUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.swing.*;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ThemeLoader {
+	private static final Logger LOG = LogManager.getLogger("Theme Loader");
+
+	private static final LinkedHashSet<Theme> THEMES = new LinkedHashSet<>();
+	public static Theme DARK_THEME;
+	public static Theme CURRENT_THEME;
+
+	public static void initUIThemes() {
+		LOG.debug("Loading UI themes");
+
+		final Gson gson = new Gson();
+
+		// Load themes
+		Set<String> files = PluginLoader.INSTANCE.getResources("themes", Pattern.compile("theme.json"));
+		for (String file : files) {
+			Theme theme = gson.fromJson(FileIO.readResourceToString(PluginLoader.INSTANCE, file), Theme.class);
+			// The ID will be used to get images from this theme if the user select it.
+			theme.setId(new File(file).getParentFile().getName());
+
+			if (theme.getColorScheme() != null) {
+				// Set the value for the Blockly Panel file
+				// If the file does not exist, we use the default file (from the default_dark them)
+				if (PluginLoader.INSTANCE.getResource(
+						"themes/" + theme.getID() + "/colors/blockly_" + theme.getColorScheme().getID() + ".css") == null) {
+					theme.getColorScheme().setBlocklyCSSFile("dark");
+					LOG.warn(theme.getColorScheme().getID()
+							+ " color scheme does not define the Blockly Panel colors! Default_dark's file will be used!");
+				} else {
+					theme.getColorScheme().setBlocklyCSSFile(theme.getColorScheme().getID());
+				}
+
+				// Set the value for the Code Editor file
+				// If the file does not exist, we use the default file (from the default_dark them)
+				if (PluginLoader.INSTANCE.getResource(
+						"themes/" + theme.getID() + "/colors/codeeditor_" + theme.getColorScheme().getID() + ".xml") == null) {
+					theme.getColorScheme().setCodeEditorFile("dark");
+					LOG.warn(theme.getColorScheme().getID()
+							+ " color theme does not define the code editor colors! Default_dark's file will be used!");
+				} else {
+					theme.getColorScheme().setCodeEditorFile(theme.getColorScheme().getID());
+				}
+			} else if (theme.getID().equals("default_dark")) {

```

Comment:
Looking at this, I think it would be better to load all themes dynamically, including dark one, which should define color scheme in JSON as other themes do.